csm/core/agent/csm_agent.py

```diff
@@ -122,10 +123,14 @@ def init():
        # Network file manager registration
         CsmRestApi._app["download_service"] = DownloadFileManager()
 
+        #Message Bus instance creation
+        CsmRestApi._app["comm_client"] = MessageBusComm()
```

Comment:
No need for this. You need to create a function in stats service call stop. In that function call comm client's stop.

Reply:
Modifications done